### PR TITLE
Rework Header section test

### PR DESCRIPTION
**Pull request by raphael**
Make test more systematic by testing each parameter independently.

### CODE REVIEW

**Comment by raphael** [comment]
@tchssk I tried to illustrate more systematic tests with this PR, it's very simple tests though. The idea is to test each argument / interesting values (e.g. single vs. multiple imports) independently. The test uses artificial / short values for the parameters to try and be consistent.

File: codegen/sections_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		s := Header(tc.Title, "testpackage", tc.Imports)
+		s.Render(buf)
+		actual := buf.String()
+		if actual != tc.Expected {
+			t.Errorf("%s: got %#v, expected %#v", k, actual, tc.Expected)
+		}
+	}
+}
+
+const (
```
Comment (tchssk) [inline codegen/sections_test.go]: These `const` should be in scope of `TestHeader()`, right? We have to take care to the duplication of that when we add some tests in this package.

**Comment by tchssk** [comment]
I understood. I left a comment in review.

File: codegen/sections_test.go (hunk 1 above)
Comment (raphael) [inline codegen/sections_test.go]: OK I'll move them in the test.

File: codegen/sections_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		title = "test title"
+	)
+	var (
+		imprt   = []*ImportSpec{&ImportSpec{Path: "test"}}
+		imports = append(imprt, &ImportSpec{Path: "other"})
+	)
+	cases := map[string]struct {
+		Title    string
+		Imports  []*ImportSpec
+		Expected string
```
Comment (tchssk) [inline codegen/sections_test.go]: These fields seem to don't need to be public.

Comment (raphael) [inline codegen/sections_test.go]: They don't but I like to do it this way to make the field names stand out when defining the test cases.

Comment (tchssk) [inline codegen/sections_test.go]: I'll write test in the same way.

**Comment by tchssk** [comment]
Now this looks good to me. Is this OK to merge?

**Comment by raphael** [comment]
Yes please!

**Pull request merged**